ql/src/java/org/apache/hadoop/hive/ql/stats/estimator/StatEstimators.java

```diff
@@ -0,0 +1,51 @@
+package org.apache.hadoop.hive.ql.stats.estimator;
+
+import java.util.Optional;
+
+import org.apache.hadoop.hive.ql.plan.ColStatistics;
+
+public class StatEstimators {
+
+  public static class WorstStatCombiner {
```

Comment:
Should this implement an interface? (I do not have a strong opinion, just a comment)

Nit. Can the name be changed? DefaultCombiner? Then you can add in the comments that it considers  max upper bounds for the properties, etc.